docs/reference/settings/ilm-settings.asciidoc

```diff
@@ -59,6 +59,15 @@ An index that was rolled over would normally match the full format,
 for example `logs-2016.10.31-000002`). 
 If the index name doesn't match the pattern, index creation fails.
 
+[[index-lifecycle-step-wait-time-threshold]]
+`index.lifecycle.step.wait_time_threshold`::
+(<<indices-update-settings,Dynamic>>, <<time-units,time value>>) 
+Time to wait while retrying a failed allocation step for the {ilm-init}
```

Comment:
@jrodewig I'm not sure if we should use "while retrying" as we are not necessarily retrying steps, but rather waiting for conditions to be fulfilled (ie. source index shards be collocated, shrunk index shards be recovered/allocated)

Reply:
I reworded to:
> Time to wait for the cluster to resolve allocation issues during an ILM `shrink` action.